agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -0,0 +1,77 @@
+package za.co.absa.atum.agent
+
+import org.apache.spark.sql.DataFrame
+import za.co.absa.atum.agent.model.{MeasureResult, Measurement}
+
+/**
+ *  AtumContext: This class provides the methods to measure Spark `Dataframe`. Also allows to add/edit/remove measures.
+ *  @param measurements: A sequences of measurements.
+ */
+
+case class AtumContext(measurements: Set[Measurement] = Set()) {
+
+  def withMeasures(
+    measure: Measurement
+  ): AtumContext =
+    this.copy(measurements = Set(measure))
+
+  def withMeasures(
+    measures: Iterable[Measurement]
+  ): AtumContext =
+    this.copy(measurements = measures.toSet)
+
+  def withMeasuresAdded(
+    measure: Measurement
+  ): AtumContext =
+    this.copy(measurements = measurements + measure)
+
+  def withMeasuresAdded(
+    measures: Iterable[Measurement]
+  ): AtumContext =
+    this.copy(measurements = measurements ++ measures)
+
+  def withMeasureRemoved(measurement: Measurement): AtumContext =
+    this.copy(measurements = measurements.filterNot(_ == measurement))
+
+}
+
+object AtumContext {
+  implicit class DatasetWrapper(df: DataFrame) {
+
+    /**
+     *  Executes the measure directly with not AtumContext.
```

Comment:
Executes the measure directly with not AtumContext.

What do you mean? Executes the measure directly outside AtumContext. ? Or when AtumContext not provided?

Reply:
In simple pipelines, you can execute many Measures without creating an `AtumContext`.